### ISSUE TITLE
Problem with device assertion

### ISSUE THREAD

**Issue by morphism90** [harvardnlp/sent-conv-torch#29]
Hi everyone, when I trying to run main.lua with my dataset, which was created due to following command:

`python preprocess_utf8.py  custom w2v/ruscorpora_russe.model.bin --train Trainset.txt --test Testset.txt --dev Validset.txt --custom_name ParentClass
`

I always get the error:
```
/tmp/luarocks_cunn-scm-1-9034/cunn/lib/THCUNN/ClassNLLCriterion.cu:52: void cunn_ClassNLLCriterion_updateOutput_kernel(Dtype *, Dtype *, Dtype *, long *, Dtype *, int, int, int, int) [with Dtype = float, Acctype = float]: block: [0,0,0], thread: [14,0,0] Assertion `t >= 0 && t < n_classes` failed.
THCudaCheck FAIL file=/tmp/luarocks_cutorch-scm-1-3543/cutorch/lib/THC/generic/THCStorage.c line=32 error=59 : device-side assert triggered
/home/ubuntu/torch/install/bin/luajit: cuda runtime error (59) : device-side assert triggered at /tmp/luarocks_cutorch-scm-1-3543/cutorch/lib/THC/generic/THCStorage.c:32
stack traceback:
        [C]: at 0x7fac9f291f60
        [C]: in function '__index'
        ...ntu/torch/install/share/lua/5.1/nn/ClassNLLCriterion.lua:52: in function 'forward'
        ./trainer.lua:187: in function 'test'
        main.lua:197: in function 'train_loop'
        main.lua:340: in function 'main'
        main.lua:353: in main chunk
        [C]: in function 'dofile'
        ...untu/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
        [C]: at 0x00405e90

```

What kind of thing caused it? In my opinion, there may be some discrepancy between sizes of dev,test, train sets. But why does it matter? It is just my hypothesis. Does anyone know something about it?

**Comment by jeffreyling** [comment]
Can you double check your test / valid set labels? They need to be 0-indexed in your .txt files - it looks like you have an out-of-bounds error.

**Comment by morphism90** [comment]
Hmmm, there is no any index in my txt files. Just first column is a label (from 0 to some number), second one and other columns are contents(features).

**Comment by morphism90** [comment]
@jeffreyling maybe there is problem in the following: I have for instance 100 examples in training set where labels are in range from 1 to 56 but in the validation set I have some labels with number outside the range( 1, 56) and it caused the error. Maybe this is the case?

**Comment by jeffreyling** [comment]
Yes, you shouldn't have classes in your val / test that aren't seen in the train. There is no way to correctly predict them, so you should omit those examples.

I'll add an assert to prevent that.
